### PR TITLE
feat(cxl-ui): cxl-vaadin-accordion theme="cxl-faq", fix #37

### PR DESCRIPTION
**Pull request by anxalicious**
Screenshot 2020-02-05 at 16 23 13

### CODE REVIEW

**Comment by lkraav** [comment]
Visuals need further upgrade, towards something like this:

image


Full fidelity accordion maybe:

image

I'm working on element style primitives that `theme=cxl-faq` can use.

File: packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { html } from 'lit-html';
+import '@conversionxl/cxl-ui/src/components/cxl-vaadin-accordion.js';
```
Comment (lkraav) [inline packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js:3]: Story needs a data file.

* Create REST API app key in your staging and/or live WP profile
* Use IntelliJ REST API client to fetch data like

```
GET https://<server>/wp-json/wp/v2/posts
Accept: application/json
Authorization: Basic anna-maria <password>
```

Modify API URL to needs, of course.

**Review by lkraav — CHANGES_REQUESTED** [review]

File: packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js (hunk 1 above)
Comment (anxalicious) [reply packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js:3]: @lkraav I got the response working but I don't know the URL there I could get the correct data

Comment (lkraav) [reply packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js:3]: Which WP REST API training have you taken?

Comment (anxalicious) [reply packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js:3]: no trainings. 
But I created the new pods in staging and got the url and data

File: packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss [hunk 2]
```diff
@@ -83,3 +83,36 @@
...
 }
+
+/**
+ * Theme "cxl-faq".
+ */
+:host([theme~="cxl-faq"]) {
+  border-bottom: 0;
+
+  [part="toggle"] {
+    margin-right: var(--lumo-space-s);
```
Comment (lkraav) [inline packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss]: Hm, previous themes seem to be using `--lumo-space-m`. Is `s` intentional here?

File: packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            </div>
+          </vaadin-accordion-panel>
+        `
+      )}
+    </cxl-vaadin-accordion>
+  `;
+};
+
+CxlVaadinAccordionThemeFaq.story = {
+  name: 'Theme: FAQ'
```
Comment (lkraav) [inline packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js]: image

Follow existing naming conventions.

File: packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss (hunk 2 above)
Comment (lkraav) [reply packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss]: After building our default styles, I now agree that design screen toggle architecture is more impactful. Let's bring back the extra style rules here, `border-radius: unset; background-color: var(--lumo-shade);` etc

**Review by lkraav — CHANGES_REQUESTED** [review]
Hm, we should probably open an upstream issue this PR fixes. Add "fix #37" to commit message.

**Comment by anxalicious** [comment]
updated the screen recording preview

File: packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss (hunk 2 above)
Comment (anxalicious) [reply packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss]: > Hm, previous themes seem to be using `--lumo-space-m`. Is `s` intentional here?

I remeber us talking about it and you said that don't put too much white space. With s, the question and answer section is vertically aligned aligned.

>  Let's bring back the extra style rules here
if we bring back black boxes, we have to reconsider s or m anyway

File: packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss [hunk 4]
```diff
@@ -101,6 +104,9 @@
 
 :host([theme~="cxl-faq"][opened]) {
   [part="toggle"] {
+    color: var(--lumo-base-color);
+    background-color: var(--lumo-primary-color);
+    border-radius: unset;
```
Comment (lkraav) [inline packages/cxl-lumo-styles/scss/themes/vaadin-accordion-panel.scss]: Oops, I was wrong, design mandates `var(--lumo-border-radius)`. Things seemed a little too blocky without.

File: packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js [hunk 5]
```diff
@@ -13,45 +15,29 @@ export const CxlVaadinAccordionThemeFaq = () => {
...
+      ${faqData.map(
+        el => html`
+          <vaadin-accordion-panel
+            id="${el.cxl_hybrid_attr_post['@attributes'].id}"
+            class="${el.cxl_hybrid_attr_post['@attributes'].class}"
+            theme="cxl-faq"
+          >
+            <header class="entry-header" slot="summary">
+              <h5 class="entry-title" itemprop="headline">
+                <a href="#">${unsafeHTML(el.title.rendered)}</a>
```
Comment (lkraav) [inline packages/storybook/cxl-ui/cxl-vaadin-accordion/theme=cxl-faq.stories.js]: FAQs aren't links. Why `<a>` tag?

**Review by lkraav — APPROVED** [review]
:guitar:

**Comment by lkraav** [comment]
OK, configure this PR to target conversionxl/aybolit:master and I'll merge it.

**Pull request merged**